src/views/Swap/components/SwapWarningModal/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+import { ModalContainer, ModalHeader, Heading, ErrorIcon, ModalBody } from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import { WrappedTokenInfo } from 'state/lists/hooks'
+import SwapWarningTokens from 'config/constants/swapWarningTokens'
+import SafemoonWarning from './SafemoonWarning'
+import BondlyWarning from './BondlyWarning'
+
+const StyledModalContainer = styled(ModalContainer)`
+  max-width: 420px;
+  border: 1px ${({ theme }) => theme.colors.warning} solid;
+`
+
+interface SwapWarningModalProps {
+  swapCurrency: WrappedTokenInfo
+  onDismiss?: () => void
+}
+
+// Modal is fired by a useEffect and doesn't respond to closeOnOverlayClick prop being set to false
```

Comment:
Not particularly happy about this - but I can't seem to get `useModal` to respond to its second argument `closeOnOverlayClick` - regardless of the value of this argument, the modal closes on overlay click. Open to suggestions here.